### PR TITLE
Added merge sort in Python

### LINKED ISSUE #3984: Merge Sort in Python
## 🚀 Feature

Category: Sorting algorithm
Topic: Merge Sort
Language: Python

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

Merge sort splits the list into two, calls itself recursively to sort both lists, and then merges the two lists into one. That means, for sufficiently large input, merge sort will be faster than bubble sort, no matter how much more efficient the bubble sort implementation is.

## Pitch

Merge sort is more efficient and works faster than quick sort in case of larger array size or datasets.

### PR DESCRIPTION
**Pull request by mehabhalodiya**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Merge sort works on the principle of Divide and Conquer. Merge sort repeatedly breaks down a list into several sublists until each sublist consists of a single element and merging those sublists in a manner that results into a sorted list.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

### CODE REVIEW

**Comment by mehabhalodiya** [comment]
@iamrajiv @HarshCasper Kindly review and merge it.

**Comment by mehabhalodiya** [comment]
@iamrajiv @HarshCasper Kindly requesting you to review and merge it.

File: Python/sort/merge_sort.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            else:
+                A[last] = B[j]
+                last, j = last - 1, j - 1
+
+        while j >= 0:
+                A[last] = B[j]
+                last, j = last - 1, j - 1
+
+if __name__ == "__main__":
+    A = [1, 3, 5, 0, 0, 0, 0]
```
Comment (iamrajiv) [inline Python/sort/merge_sort.py]: @mehabhalodiya Take input from the user.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: Python/sort/merge_sort.py (hunk 1 above)
Comment (mehabhalodiya) [reply Python/sort/merge_sort.py:48]: Okay, I am removing class and then defining functions.
Further will take input from user.

Comment (mehabhalodiya) [reply Python/sort/merge_sort.py:48]: As with `class`, it will get more complicated.

File: Python/sort/merge_sort.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Purpose:
+It works on the principle of Divide and Conquer. Merge sort repeatedly breaks
+down a list into several sublists until each sublist consists of a single
+element and merging those sublists in a manner that results into a sorted list.
+
+Methodology:
+Given two sorted integer arrays A and B, merge B into A as one sorted array.
+
+Time Complexity:  O(n)
+Space Complexity: O(1)
```
Comment (ankitaggarwal23) [inline Python/sort/merge_sort.py]: - Add Time and Space Complexity at the end of the Code

File: Python/sort/merge_sort.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+class Solution:
+    """
+    @param A  a list of integers
+    @param m  an integer, length of A
+    @param B  a list of integers
+    @param n  an integer, length of B
+    @return nothing
+    """
+
```
Comment (ankitaggarwal23) [inline Python/sort/merge_sort.py]: Remove the redundant comments

File: Python/sort/merge_sort.py (hunk 1 above)
Comment (ankitaggarwal23) [reply Python/sort/merge_sort.py]: @mehabhalodiya Take dynamic input from user don't use static inputs

File: Python/sort/merge_sort.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        while j >= 0:
+                A[last] = B[j]
+                last, j = last - 1, j - 1
+
+if __name__ == "__main__":
+    A = [1, 3, 5, 0, 0, 0, 0]
+    B = [2, 4, 6, 7]
+    Solution().merge(A, 3, B, 4)
+    print A
```
Comment (ankitaggarwal23) [inline Python/sort/merge_sort.py]: - Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by mehabhalodiya** [comment]
Yes, I am doing all suggested changes and writing new code as I am messing up modifying this same script.

**Comment by mehabhalodiya** [comment]
@iamrajiv @ankitaggarwal23 Kindly review this.

**Comment by mehabhalodiya** [comment]
@iamrajiv @ankitaggarwal23 Kindly review this.
I have made suggested changes.

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here](https://www.sanfoundry.com/python-program-implement-merge-sort/).

**Comment by mehabhalodiya** [comment]
Hello @ankitaggarwal23 I have not copied it.
Kindly requesting you to open and I will do required changes, if you want.

Suggest me and then you can remove the label of plagiarism.

**Pull request closed without merging**